test/built-ins/Temporal/PlainDate/prototype/toString/order-of-operations.js

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plaindate.prototype.tostring
+description: Properties on an object passed to toString() are accessed in the correct order
+includes: [compareArray.js, temporalHelpers.js]
+features: [Temporal]
+---*/
+
+const expected = [
+  "get options.calendarName",
+  "get options.calendarName.toString",
+  "call options.calendarName.toString",
+  "get this.calendar[Symbol.toPrimitive]",
```

Comment:
Not to be a pain in the butt, but if we want to test every possible observation, we could ensure calendar's Symbol.toPrimitive method is called as well, and that there's a toString call on the result. In general, for my education on test262, what's the threshold for how in-depth to go here? To me checking all these calls could be pretty excessive, but I don't know what bar to set for test262.

Reply:
There could certainly be different opinions on this, but to me that doesn't add a lot of value. Get @@toPrimitive followed by Get toString and Call toString, on an object with no `[Symbol.toPrimitive]` property, most likely indicates that ToString is used correctly here.

To be entirely exhaustive here you'd have to check many more cases than that (e.g., valueOf but no toString, @@toPrimitive returns an object with @@toPrimitive up to an arbitrary depth), and I don't think it provides any clear benefit to implementors or web compatibility.

Reply:
Thank you, that's a good explanation. As someone new to test262, in my tests for Array.fromAsync, I find myself wondering how deep down the rabbit hole of checking iterator protocol calls to go - I'll try to use this as a guideline to remove some of the extra-deep checks :)